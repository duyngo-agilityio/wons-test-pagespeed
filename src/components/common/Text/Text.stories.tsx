import type { Meta, StoryObj } from '@storybook/react';

// components
import Text from './index';

const meta: Meta<typeof Text> = {
  title: 'Components/Common/Text',
  component: Text,
  parameters: {
    layout: 'centered',
    controls: { expanded: true },
  },
  tags: ['autodocs'],
  argTypes: {
    as: {
      description: 'HTML tag',
      control: { type: 'select' },
      options: ['p', 'span', 'div'],
    },
    className: {
      description: 'Additional class name',
      control: 'text',
    },
  },
} satisfies Meta<typeof Text>;

export default meta;

type Story = StoryObj<typeof Text>;

export const TextStory: Story = {
  args: {
    className: 'text-[30px]',
    children:
      ' Lorem, ipsum dolor sit amet consectetur adipisicing elit. Consectetur',
  },
};
