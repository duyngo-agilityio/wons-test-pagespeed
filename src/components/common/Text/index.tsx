// Libs
import { memo } from 'react';

interface TextProps {
  text: string;
  as?: keyof React.ReactHTML;
  className?: string;
}

const Text = ({ text, as = 'p', className = '' }: TextProps) => {
  const Component = as;

  return (
    <Component className={`text-3xl font-dm-sans dark:text-white ${className}`}>
      {text}
    </Component>
  );
};

export default memo(Text);
