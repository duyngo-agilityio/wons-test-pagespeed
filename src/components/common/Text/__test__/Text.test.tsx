import { render } from '@testing-library/react';

// components
import Text from '../index';

describe('Text Component', () => {
  it('renders correctly with all custom props', () => {
    const component = render(
      <Text className="text-[40px]">
        Lorem, ipsum dolor sit amet consectetur adipisicing elit. Consectetur,
        veniam doloribus dicta fuga sed soluta blanditiis sequi porro corrupti
        voluptatum adipisci, dolorum, quae fugiat atque? Aliquid dolorum modi
        architecto sapiente!
      </Text>,
    );
    expect(component).toMatchSnapshot();
  });
});
