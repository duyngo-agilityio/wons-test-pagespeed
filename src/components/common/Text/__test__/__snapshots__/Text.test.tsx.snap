// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Component renders correctly with all custom props 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <p
        class="text-3xl font-dm-sans dark:text-white text-[40px]"
      >
         Lorem, ipsum dolor sit amet consectetur adipisicing elit. Consectetur,   veniam doloribus dicta fuga sed soluta blanditiis sequi porro corrupti   voluptatum adipisci, dolorum, quae fugiat atque? Aliquid dolorum modi   architecto sapiente!
      </p>
    </div>
  </body>,
  "container": <div>
    <p
      class="text-3xl font-dm-sans dark:text-white text-[40px]"
    >
       Lorem, ipsum dolor sit amet consectetur adipisicing elit. Consectetur,   veniam doloribus dicta fuga sed soluta blanditiis sequi porro corrupti   voluptatum adipisci, dolorum, quae fugiat atque? Aliquid dolorum modi   architecto sapiente!
    </p>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
